src/printer.js

```diff
@@ -4322,9 +4349,19 @@ function printArgumentsList(path, options, print) {
   const closingLinebreakAnyway =
     parensUnnecessary && parent.type === 'JSXExpressionContainer'
 
+  let tmpNeedsParens
   const openingImplicitSpace =
     args.length === 1 &&
-    path.call(argPath => pathNeedsParens(argPath, options), 'arguments', '0')
+    path.call(
+      argPath =>
+        (tmpNeedsParens = pathNeedsParens(argPath, options)) &&
+        !(
+          tmpNeedsParens.unlessParentBreaks &&
+          tmpNeedsParens.unlessParentBreaks.visibleType
```

Comment:
This refinement of #76 results in cases where a space is left between a callee and its only parenthesized arg (eg see below tests eg:)
```
d = if true then id (if false then undefined else nonce)
```
as opposed to
```
d = if true then id(if false then undefined else nonce)
```

Which I think is not a big deal. The inconsistency isn't great. But I find that rule (of always trying to "remove the space" between an implicit callee and its only parenthesized argument) a little questionable anyway